### PR TITLE
Reference Book

### PR DESCRIPTION
**Pull request by karenc**
- Add uuid and version fields to content pages
- Store uuid and version when importing cnx pages
- Add pages API to return content html for a cnx uuid
- Generate random uuid and version for content page factory
- Add cnx_id to pages returned by readings API

### CODE REVIEW

File: app/subsystems/content/routines/create_page.rb [hunk 1]
```diff
@@ -4,12 +4,15 @@ class Content::Routines::CreatePage
 
   protected
 
-  def exec(url:, title:, content:, book_part:, chapter_section:)
+  def exec(url:, title:, content:, book_part:, chapter_section:, id:)
```
Comment (jpslav) [inline app/subsystems/content/routines/create_page.rb]: I'm not a big fan of passing in an `id` field to this routine as it seems like it lets the caller set the primary key.  Can we just pass in the `uuid` and `version` and do the split in `ImportPage`?

**Comment by karenc** [comment]
Ready for review.  Changes to links in the content html will be in another PR.

**Comment by philschatz** [comment]
@dak thoughts?

**Comment by jpslav** [comment]
Looks good to me.  Can merge when @dak or @philschatz weigh in (or they can just give the go ahead and we can fix issues in followup PR)

**Comment by philschatz** [comment]
@dak's out and I'm fine with a followup PR

**Pull request merged**